src/configSchema.ts

```diff
@@ -0,0 +1,167 @@
+	switch (error.code) {
+		case z.ZodIssueCode.invalid_union:
+			return {
+				message: error.unionErrors
+					.reduce((acc, error) => {
+						// @ts-expect-error avoids type conflict string to never
+						error.errors.forEach((x) => acc.push(x.message));
+						return acc;
+					}, [])
+					.join("; "),
+			};
+	}
+
+	return { message: ctx.defaultError };
+};
+/**
+ * helper functions for validation
+ * @return transformed duration (string -> milliseconds)
+ */
+
+function transformDurationString(durationStr: string, ctx) {
+	const duration = ms(durationStr);
+	if (isNaN(duration)) {
+		// adds the error to the Zod Issues
+		ctx.addIssue({
+			code: "custom",
+			message: zodErrorMsg.vercel,
+		});
+	}
+	return duration;
+}
+
+/**
+ * an object of the zod schema
+ * each are named after what they are intended to validate
+ */
+
+export const VALIDATION_SCHEMA = z
+	.object({
+		delay: z
+			.number()
+			.positive({
+				message: zodErrorMsg.delay,
+			})
+			.default(10),
+		torznab: z.array(z.string().url()),
+		dataDirs: z.array(z.string()).nullish(),
+		matchMode: z.nativeEnum(MatchMode),
+		dataCategory: z.string().nullish(),
+		linkDir: z.string().nullish(),
+		linkType: z.nativeEnum(LinkType),
+		skipRecheck: z.boolean(),
+		maxDataDepth: z.number().gte(1),
+		torrentDir: z.string(),
+		outputDir: z.string(),
+		includeEpisodes: z.boolean(),
+		includeSingleEpisodes: z.boolean(),
+		includeNonVideos: z.boolean(),
+		fuzzySizeThreshold: z.number().positive().lte(1, {
+			message: zodErrorMsg.fuzzySizeThreshold,
+		}),
+		excludeOlder: z
+			.string()
+			.min(1, { message: zodErrorMsg.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		excludeRecentSearch: z
+			.string()
+			.min(1, { message: zodErrorMsg.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		action: z.nativeEnum(Action),
+		qbittorrentUrl: z.string().url().nullish(),
+		rtorrentRpcUrl: z.string().url().nullish(),
+		transmissionRpcUrl: z.string().url().nullish(),
+		delugeRpcUrl: z.string().url().nullish(),
+		duplicateCategories: z.boolean(),
+		notificationWebhookUrl: z.string().url().nullish(),
+		port: z
+			.number()
+			.positive()
+			.lte(65535)
+			.or(z.boolean().transform(() => null)),
+		host: z.string().ip().nullish(),
+		rssCadence: z
+			.string()
+			.min(1, { message: zodErrorMsg.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		searchCadence: z
+			.string()
+			.min(1, { message: zodErrorMsg.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		snatchTimeout: z
+			.string()
+			.min(1, { message: zodErrorMsg.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		searchTimeout: z
+			.string()
+			.min(1, { message: zodErrorMsg.emptyString })
+			.transform(transformDurationString)
+			.nullish(),
+		searchLimit: z.number().positive().nullish(),
+		apiAuth: z.boolean().default(false),
+		verbose: z.boolean().default(false),
```

Comment:
To elaborate, apiAuth I think we are forcing right? So that needs to be addressed as well (remove the option and force true, I can just do that with Zod I think or we can hardcode it. The thing with verbose is that it's not in the config template by default - so we'd be relying on fallback in commander and just strictly forcing a boolean.